### PR TITLE
Add HiveHadoop2 connector

### PR DESCRIPTION
**Pull request by majetideepak**
This PR adds the HiveHadoop2 connector

It extends the exisiting HiveConnector

### CODE REVIEW

**Comment by kgpai** [comment]
Hi Deepak, 
I am working on merging this today. Will try and get it merged today.

**Review by kgpai — APPROVED** [review]

File: velox/connectors/hive/HiveConnector.h [hunk 1]
```diff
@@ -29,6 +29,11 @@
 
 namespace facebook::velox::connector::hive {
 
+namespace {
+const std::string kHiveConnectorName = "hive";
```
Comment (kgpai) [inline velox/connectors/hive/HiveConnector.h:33]: Is there a reason to have these here ; they seem to be only used in 1 place each, why not make them default values ?

Comment (majetideepak) [reply velox/connectors/hive/HiveConnector.h:33]: This variable is used in one of the test as well. This is refactored from another header file.

Comment (majetideepak) [reply velox/connectors/hive/HiveConnector.h:33]: I will probably make them class members.

**Comment by kgpai** [comment]
Update : This should be merged by today.

**Comment by kgpai** [comment]
Update: This is delayed due to a clang diagnostics bug which we are fixing and hope to get resolved soon. I will post another update by EOD.

**Pull request closed without merging**